src/file/dto/file.dto.ts

```diff
@@ -0,0 +1,36 @@
+import { FilterableField, IDField } from "@nestjs-query/query-graphql";
+import { ObjectType, ID, GraphQLISODateTime, Field } from "@nestjs/graphql";
+import { UPLOAD_TYPE } from "src/constants";
+
+@ObjectType("file")
+export class FileDTO {
+  @IDField(() => ID)
+  id!: string;
+
+  @FilterableField()
+  filename!: string;
+
+  @Field()
+  mimetype!: string;
+
+  @Field({ nullable: true })
+  filenameOnBucket: string;
+
+  @FilterableField()
+  uploadingStatus!: UPLOAD_TYPE;
+
+  @Field({ nullable: true })
+  size: number;
+
+  @Field({ nullable: true })
+  reasonOfFailure: string;
+
+  @Field({ nullable: true })
+  downloadLink: string;
```

Comment:
Are we using this field `downloadLink`?

Reply:
Yes, if the file is successfully uploaded to our bucket then we can get the URL to hit and it will start downloading that file else returns null. Something looks like bellow

```
{
   ...response,
   downloadLink: http://localhost:3000/download/{fileId}
}

```

Reply:
This can be generated real-time on client side as well. Doing this is adding a extra iteration in assembler.

Reply:
commit - [83f94b1](https://github.com/parthc2fo/upload-download-thing/pull/4/commits/83f94b116562b8b5b7baa6ad0dc4460514c734b8)